test/e2e/mocked_agent.go

```diff
@@ -0,0 +1,198 @@
+package e2e
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	sdkEvent "github.com/newrelic/infra-integrations-sdk/data/event"
+	"github.com/newrelic/nri-kube-events/pkg/events"
+	"github.com/newrelic/nri-kube-events/pkg/sinks"
+	log "github.com/sirupsen/logrus"
+)
+
+// Must be in sync with unexported name in pkg/sinks/new_relic_infra.go:32.
+const newRelicInfraSinkID = "newRelicInfra"
+
+// MockedAgentSink is an instrumented infra-agent sink for testing e2e reception and processing.
+type MockedAgentSink struct {
+	agentSink         events.Sink
+	httpServer        *httptest.Server
+	eventReceivedChan chan struct{}
+	receivedEvents    []sdkEvent.Event
+	mtx               *sync.RWMutex
+}
+
+// NewMockedAgentSink returns an instrumented infra-agent sink for testing.
+func NewMockedAgentSink() *MockedAgentSink {
+	mockedAgentSink := &MockedAgentSink{
+		mtx:               &sync.RWMutex{},
+		eventReceivedChan: make(chan struct{}, 128),
+	}
+	mockedAgentSink.httpServer = httptest.NewServer(mockedAgentSink)
+
+	agentSinkConfig := sinks.SinkConfig{
+		Name: newRelicInfraSinkID,
+		Config: map[string]string{
+			"clusterName":   "integrationTest",
+			"agentEndpoint": "http://" + mockedAgentSink.httpServer.Listener.Addr().String(),
+		},
+	}
+
+	createdSinks, err := sinks.CreateSinks([]sinks.SinkConfig{agentSinkConfig})
+	if err != nil {
+		log.Fatalf("error creating infra sink: %v", err)
+	}
+
+	agentSink, ok := createdSinks[newRelicInfraSinkID]
+	if !ok {
+		log.Fatal("could not retrieve agent infra sink from map")
+	}
+
+	mockedAgentSink.agentSink = agentSink
+
+	return mockedAgentSink
+}
+
+// HandleEvent sends a notification to the event received channel and then forwards it to the underlying sink.
+func (mas *MockedAgentSink) HandleEvent(kubeEvent events.KubeEvent) error {
+	mas.eventReceivedChan <- struct{}{}
+	return mas.agentSink.HandleEvent(kubeEvent)
+}
+
+// ServeHTTP handles a request that would be for the infra-agent and stores the unmarshalled event.
+func (mas *MockedAgentSink) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
```

Comment:
Usually just `w` is used.
```suggestion
func (mas *MockedAgentSink) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Reply:
I'm a strong detractor of `w`, not on my watch :P